### PR TITLE
[DataLoader] Make all resolve functions agnostic about how data is fetched.

### PR DESCRIPTION
**Pull request by alloy**
- Per HTTP-request fresh DataLoader instances
- Refactor to make future batching possible

### CODE REVIEW

**Comment by mzikherman** [comment]
Pushed another commit which switches _most_ of the artist resolves to use individual loaders per type (+ positron).

The remaining loaders used in the artist resolves are [total](https://github.com/artsy/metaphysics/blob/1bf01a296a220dffc3f3982b75d2ce2e79a650c9/lib/loaders/total.js) (which I think are just special cases of the existing loaders), and also a tricksy [numeral](https://github.com/artsy/metaphysics/blob/1bf01a296a220dffc3f3982b75d2ce2e79a650c9/schema/fields/numeral.js#L8-L19) generic field.

**Comment by alloy** [comment]
Alright, I think this is good for a first merge.

File: lib/loaders/per_type.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import positron from '../apis/positron';
+import { toKey } from '../helpers';
+import httpLoader from './http';
+
+// TODO As we’re currently re-using the http loader as it was, it won’t support batching yet.
+//      We should rework that once we’ve changed all resolve functions to only use per-type loaders.
+const apiLoader = (api, path) => {
+  const loader = httpLoader(api);
+  return (id, params) => {
+    const key = toKey(typeof(path) === 'function' ? path(id) : path, params);
```
Comment (mzikherman) [inline lib/loaders/per_type.js]: 👍 cool

**Comment by mzikherman** [comment]
hmm ❌

**Comment by mzikherman** [comment]
Nice, let's go for it!

Safe to merge + deploy! The artist schema isn't _fully_ migrated tho, there's still the `total` and `numeral` stuff, will take a look.

Either way, incremental commits/merges/deploys ftw

**Pull request merged**